docs/stdlib/fts.rst

```diff
@@ -0,0 +1,575 @@
+      },
+    }
+
+Finally, the search query supports features for fine-tuning (5). By default,
+all search terms are desirable, but ultimately optional. You can enclose a
+term or even a phrase in ``"..."`` to indicate that this specific term is of
+increased importance and should appear in all matches:
+
+.. code-block:: edgeql-repl
+
+    edgedb> with res := (
+    .......   select fts::search(
+    .......     Item, '"corn sugar"',
+    .......     language := 'eng',
+    .......   )
+    ....... )
+    ....... select res.object {name, description, score := res.score}
+    ....... order by res.score desc;
+    {
+      default::Item {
+        name: 'Sweet',
+        description: 'Treat made with corn sugar',
+        score: 0.4955161,
+      },
+    }
+
+Only one ``Item`` contains the phrase "corn sugar" and incomplete matches are
+omitted.
+
+The search query can also use ``AND`` (using upper-case to indicate that it is
+a query modifier and not part of the query) to indicate whether terms are
+required or optional:
+
+.. code-block:: edgeql-repl
+
+    edgedb> with res := (
+    .......   select fts::search(
+    .......     Item, 'sweet AND treat',
+    .......     language := 'eng',
+    .......   )
+    ....... )
+    ....... select res.object {name, description, score := res.score}
+    ....... order by res.score desc;
+    {
+      default::Item {
+        name: 'Sweet',
+        description: 'Treat made with corn sugar',
+        score: 0.70076555,
+      },
+    }
+
+Finally, adding a ``!`` in front of a search term marks it as something that
+the matching object *must not* contain:
+
+.. code-block:: edgeql-repl
+
+    edgedb> with res := (
+    .......   select fts::search(
+    .......     Item, '!treat',
+    .......     language := 'eng',
+    .......   )
+    ....... )
+    ....... select res.object {name, description, score := res.score}
+    ....... order by res.score desc;
+    {
+      default::Item {
+        name: 'Canned corn',
+        description: {},
+        score: 0,
+      },
+    }
+
+
+----------
+
+
+.. eql:type:: fts::Language
+
+    An :eql:type:`enum` for representing commonly supported languages.
+
+    When indexing an object for full-text search it is important to specify
+    the expected language by :eql:func:`with_options` function. This
+    particular :eql:type:`enum` represents languages that are common across
+    several possible [future] backend implementations and thus are "safe" even
+    if the backend implementation switches from one of the options to another.
+
+    The following ISO 639-3 language identifiers are available:
+
+    * ``ara``
+    * ``hye``
+    * ``eus``
+    * ``cat``
+    * ``dan``
+    * ``nld``
+    * ``eng``
+    * ``fin``
+    * ``fra``
+    * ``deu``
+    * ``ell``
+    * ``hin``
+    * ``hun``
+    * ``ind``
+    * ``gle``
+    * ``ita``
+    * ``nor``
+    * ``por``
+    * ``ron``
+    * ``rus``
+    * ``spa``
+    * ``swe``
+    * ``tur``
+
```

Comment:
Maybe link to the list of language codes: https://iso639-3.sil.org/code_tables/639/data?field_iso639_cd_st_mmbrshp_639_1_tid=94671